src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -178,6 +190,56 @@ public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String d
         }
     }
 
+    public DRSAccessURL getAccessUrlForObjectId(AuthenticatedUserRequest authUser, String objectId, String accessId) {
+        DRSObject object = lookupObjectByDrsId(authUser, objectId, false);
+
+        DrsId drsId = drsIdService.fromObjectId(objectId);
+        Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
+        SnapshotProject snapshotProject = snapshotService.retrieveAvailableSnapshotProject(snapshot.getId());
+        GoogleProjectResource projectResource = snapshot.getProjectResource();
+
+        List<DRSAccessMethod> matchingAccessMethods = getAccessMethodsMatchingAccessId(accessId, object);
+        if (matchingAccessMethods.size() > 0) {
+            Storage storage = StorageOptions.newBuilder().setProjectId(projectResource.getGoogleProjectId()).build().getService();
+            String urlString = matchingAccessMethods.get(0).getAccessUrl().toString();
+            Pattern gsBucketRegex = Pattern.compile("(?<=//)(.*?)(?=/)");
```

Comment:
I'm thinking we may want to have the logic of pulling out the bucket in the `StorageUtils` class

Reply:
Oh actually check out: bio.terra.service.filedata.google.gcs.GcsPdao#getGcsLocatorFromGsPath. Looks like there's a method that pulls out a bucket from a gs path for you already!